proof/verifier_test.go

```diff
@@ -0,0 +1,613 @@
+	require.NoError(t, err)
+
+	err = c.Upsert(stxoCommitment)
+	require.NoError(t, err)
+
+	// Before we create proofs, add any additional STXO assets to the
+	// commitment.
+	if cfg.addAdditionalStxoProofs {
+		for i := 1; i < len(p.Asset.PrevWitnesses); i++ {
+			spentAsset := makeSpentAsset(
+				t, p.Asset.PrevWitnesses[i],
+			)
+
+			err := stxoCommitment.Upsert(spentAsset)
+			require.NoError(t, err)
+
+			err = c.Upsert(stxoCommitment)
+			require.NoError(t, err)
+		}
+	}
+
+	_, stxoProof0, err := c.Proof(
+		asset.EmptyGenesisID, spentAsset0.AssetCommitmentKey(),
+	)
+	require.NoError(t, err)
+
+	_, v0Proof, err := c.Proof(
+		p.Asset.TapCommitmentKey(), p.Asset.AssetCommitmentKey(),
+	)
+	require.NoError(t, err)
+	p.InclusionProof.CommitmentProof.Proof = *v0Proof
+
+	cp := p.InclusionProof.CommitmentProof
+	if cfg.addStxoProof {
+		cp.STXOProofs = map[asset.SerializedKey]commitment.Proof{
+			key0: *stxoProof0,
+		}
+	}
+
+	if cfg.addAdditionalStxoProofs {
+		for i := 1; i < len(p.Asset.PrevWitnesses); i++ {
+			spentAsset := makeSpentAsset(
+				t, p.Asset.PrevWitnesses[i],
+			)
+			key := asset.ToSerialized(spentAsset.ScriptKey.PubKey)
+
+			_, stxoProof, err := c.Proof(
+				asset.EmptyGenesisID,
+				spentAsset.AssetCommitmentKey(),
+			)
+			require.NoError(t, err)
+
+			cp.STXOProofs[key] = *stxoProof
+		}
+	}
+
+	if cfg.updateAnchorTx {
+		p.AnchorTx.TxOut[0].PkScript = outputScript(
+			t, c, p.InclusionProof.InternalKey,
+		)
+	}
+
+	return *stxoProof0, key0
+}
+
+func addV0ExclusionOutput(t *testing.T, p *Proof, c *commitment.TapCommitment,
+	tapKey, assetKey [32]byte, internalKey *btcec.PublicKey,
+	outputIndex uint32) {
+
+	_, exclusionProof, err := c.Proof(tapKey, assetKey)
+	require.NoError(t, err)
+
+	p.AnchorTx.TxOut = append(p.AnchorTx.TxOut, &wire.TxOut{
+		PkScript: outputScript(t, c, internalKey),
+	})
+	p.ExclusionProofs = []TaprootProof{{
+		OutputIndex: outputIndex,
+		InternalKey: internalKey,
+		CommitmentProof: &CommitmentProof{
+			Proof: *exclusionProof,
+		},
+	}}
+}
+
+func addV1ExclusionProof(t *testing.T, p *Proof, c *commitment.TapCommitment,
+	outputIndex uint32) {
+
+	spentAsset := makeSpentAsset(t, p.Asset.PrevWitnesses[0])
+	key := asset.ToSerialized(spentAsset.ScriptKey.PubKey)
+
+	_, stxoExclusionProof, err := c.Proof(
+		asset.EmptyGenesisID, spentAsset.AssetCommitmentKey(),
+	)
+	require.NoError(t, err)
+
+	for idx := range p.ExclusionProofs {
+		if p.ExclusionProofs[idx].OutputIndex != outputIndex {
+			continue
+		}
+
+		cp := p.ExclusionProofs[idx].CommitmentProof
+		cp.STXOProofs = map[asset.SerializedKey]commitment.Proof{
+			key: *stxoExclusionProof,
+		}
+	}
+}
+
+// TestVerifyV0V1InclusionProofs tests that we properly verify the inclusion
+// proofs for both v0 and v1.
+func TestVerifyV1InclusionProof(t *testing.T) {
+	testCases := []struct {
+		name                string
+		makeProof           func(t *testing.T) *Proof
+		expectedErr         error
+		expectedErrContains string
+	}{{
```

Comment:
Style nit: can put the frist case on a new line, then each section after that adds a new test, so like: 
```go
t = []struct {
}{
  case
}, 
{
  case
}, 
}
  case
}, 
```

Reply:
I'm going to vote in favor of less indentation here, to prevent line wrapping hell.